AlphaGo/training/supervised_policy_trainer.py

```diff
@@ -411,331 +367,294 @@ def load_train_val_test_indices(verbose, arg_symmetries, dataset_length, batch_s
-    # warning mentioning accuracy from keras
-    if len(train_indices) % batch_size != 0:
-        # remove first len(train_indices) % args.minibatch rows
-        train_indices = np.delete(train_indices, [row for row in range(len(train_indices)
-                                                  % batch_size)], 0)
-
-    if verbose:
-        print("dataset loaded")
-        print("\t%d total positions" % dataset_length)
-        print("\t%d total samples" % (dataset_length * len(symmetries)))
-        print("\t%d total samples check" % (len(train_indices) +
-              len(val_indices) + len(test_indices)))
-        print("\t%d training samples" % len(train_indices))
-        print("\t%d validation samples" % len(val_indices))
-        print("\t%d test samples" % len(test_indices))
+    # Need to make sure training set size is divisible by minibatch size for the batch generator.
+    n_remove = len(train_indices) % batch_size
+    train_indices = train_indices[n_remove:]
 
     return train_indices, val_indices, test_indices
 
 
-def set_training_settings(resume, args, metadata, dataset_length):
-    """ save all args to metadata,
-        check if critical settings have been changed and prompt user about it.
-        create new shuffle files if needed.
+def validate_args(args):
+    """Save all args to metadata, check if critical settings have been changed and prompt user about
+       it. Creates new shuffle files if needed.
+
+       Returns (metdata, resume_weights)
     """
+    # Resuming with defaults if args.mode is 'resume', or with sanity-checks on all settings if
+    # in 'train' mode but 'weights' are set.
+    resume = args.mode == 'resume' or args.weights is not None
 
-    # shuffle file locations for train/validation/test set
+    # Shuffle file locations for train/validation/test set.
     shuffle_file_train = os.path.join(args.out_directory, FILE_TRAIN)
-    shuffle_file_val = os.path.join(args.out_directory, FILE_VALIDATE)
     shuffle_file_test = os.path.join(args.out_directory, FILE_TEST)
+    shuffle_file_val = os.path.join(args.out_directory, FILE_VALIDATE)
 
-    # determine if new shuffle files have to be created
+    # Determine if new shuffle files have to be created.
     save_new_shuffle_indices = not resume
 
-    # save current symmetries to metadata
-    metadata["symmetries"] = args.symmetries
-
     if resume:
-        # check if argument model and meta model are the same
-        if metadata["model_file"] != args.model:
-            # verify if user really wants to use new model file
-            print("the model file is different from the model file used last run: " +
-                  metadata["model_file"] + ". It might be different than the old one.")
-            if args.override or not confirm("Are you sure you want to use the new model?", False):  # noqa: E501
-                raise ValueError("User abort after mismatch model files.")
```

Comment:
i decided to assume typo error (in most cases) when people select no and to use value stored in metadata.
i think using data from metadata is more user friendly as they do not have to correct their typo.